tests/gold_tests/redirect/redirect_actions.test.py

```diff
@@ -0,0 +1,225 @@
+'''
+Test redirection behavior to invalid addresses
+'''
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from enum import Enum
+import re
+import os
+import socket
+Test.Summary = '''
+Test redirection behavior to invalid addresses
+'''
```

Comment:
We may want to add:
```Python
Test.SkipIf(Condition.true('autest sometimes does not capture output on the last test case of a scenario'))
```

There seems to be an issue in autest with some buffered I/O. A significant precentage of the time, after tcp_client has flushed the bytes it received to stdout, autest shows failure with no output. I've observed this in Apache CI as well as my local CentOS7 environment. This happens only on the last test case in a scenario, before the Trafficserver under test is stopped. If it were a problem with the test itself, we would expect to see it happen on all test cases. So I think autest is missing a read or not flushing a write on the last time through a loop. I did not have time to look, and I would rather not enable an automated test with false failures some of the time. I used these tests to verify the correctness of the changes manually, so I think they are valuable to keep so that we can enabled them if/when we find and fix the problem.

Reply:
I think this has been found to be an AuTest issue - Jesse is tracking it down.

Reply:
Planning to add the skip until we can figure out what is wrong.